modules/fun.js

```diff
@@ -0,0 +1,450 @@
+    default:
+    case (`icarus`): {
+      const aiChoice = u.rand(Object.keys(hbsValues));
+      int.editReply(`**Handicorn, Buttermelon, Sloth, Fight!**\n` +
+      chooser + ` challenged Icarus!\n` +
+      hbsResult(chooser, choice, `Icarus`, aiChoice));
+    }
+  }
+  /**
+ * function hbsResult
+ * @param {string} chooser1 a string to represent who made choice 1
+ * @param {string} choice1 chooser1's `Handicorn`, `Buttermelon`, or `Sloth` choice
+ * @param {string} chooser2 ...
+ * @param {string} choice2 ...
+ * @return {string} a summary including who picked what and who won.
+ */
+  function hbsResult(chooser1, choice1, chooser2, choice2) {
+    let response = `${chooser1} picked ${hbsValues[choice1].emoji}, ${chooser2} picked ${hbsValues[choice2].emoji}.\n`;
+    const diff = hbsValues[choice2].value - hbsValues[choice1].value;
+    if (diff == 0) {
+      response += `It's a tie!`;// TIE
+    } else if ((diff == -1) || (diff == 2)) {
+      response += `${chooser2} wins!`;
+    } else {
+      response += `${chooser1} wins!`;
+    }
+    return response;
+  }
+}
+/**
+ * function slashFunAcronym
+ * @param {Discord.ChatInputCommandInteraction} int a /fun acronym interaction
+ */
+async function slashFunAcronym(int) {
+  const alphabet = [`A`, `B`, `C`, `D`, `E`, `F`, `G`, `H`, `I`, `J`, `K`, `L`, `M`, `N`, `O`, `P`, `Q`, `R`, `S`, `T`, `U`, `V`, `W`, `Y`, `Z`];
+  const len = int.options.getInteger(`length`) || Math.floor(Math.random() * 3) + 3;
+  const profanityFilter = require(`profanity-matcher`);
+  const pf = new profanityFilter();
+  let wordgen = [];
+
+  for (let ignored = 0; ignored < len * len; ignored++) {// try a bunch of times
+    for (let i = 0; i < len; i++) {
+      wordgen.push(alphabet[Math.floor(Math.random() * alphabet.length)]);
+    }
+    const word = wordgen.join(``);
+
+    if (pf.scan(word.toLowerCase()).length == 0) {
+      return int.editReply(`I've always wondered what __**${word}**__ stood for...`);
+    } else {
+      wordgen = [];
+    }
+  }
+  return int.editReply(`I ran into an error, try again and/or ping a botadmin.`);
+}
+
+/**
+ * function slashFunMinesweeper
+ * @param {Discord.ChatInputCommandInteraction} int a /fun minesweeper interaction
+ */
+async function slashFunMinesweeper(int) {
+  let size, mineCount;
+  // difficulty values
+  switch (int.options.getString(`difficulty`)) {
+    case `Hard`:
+      size = 14;
+      mineCount = 60;
+      break;
+    case `Medium`:
+      size = 10;
+      mineCount = 30;
+      break;
+    default:
+    case `Easy`:
+      size = 5;
+      mineCount = 5;
+      break;
+  }
+  // generate the minefield
+  let field = "";
+  // Getting all possible board spaces
+  const possibleSpaces = Array.from({ length: size * size }, (v, k) => k);
+  // Remove 4 corners, corners can't be mines
+  possibleSpaces.splice((size * size) - 1, 1);
+  possibleSpaces.splice((size - 1) * size, 1);
+  possibleSpaces.splice(size - 1, 1);
+  possibleSpaces.splice(0, 1);
```

Comment:
Might have to test this, but you might be able to chain these (ie `possibleSpaces.splice().splice()`)

Reply:
most of the mines function is just copied and I don't have a very good understanding of what it is actually doing... I'm not even sure what splice does at the moment.

Reply:
the entire algorithm got reworked. If you approve of the current state of the algorithm then this may be resolved.

Reply:
we've completely reworked the algorithm at this point... lets start a new discussion if what we have is not to be the final algorithm.